### PR TITLE
refactor(abigen): keep and use parsed spans

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Keep spans around instead of unparsing into Strings

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Updated the changelog
-   [ ] Breaking changes

### CODE REVIEW

File: ethers-contract/ethers-contract-abigen/src/lib.rs [hunk 1]
```diff
@@ -66,34 +66,34 @@ use std::{collections::HashMap, fmt, fs, io, path::Path};
...
+    pub method_aliases: HashMap<String, String>,
 
     /// Manually specified event name aliases.
-    event_aliases: HashMap<String, String>,
+    pub event_aliases: HashMap<String, String>,
 
     /// Manually specified error name aliases.
-    error_aliases: HashMap<String, String>,
+    pub error_aliases: HashMap<String, String>,
+
```
Comment (mattsse) [inline ethers-contract/ethers-contract-abigen/src/lib.rs:85]: why do we need pub access now?

**Review by mattsse — CHANGES_REQUESTED** [review]
looks great.

but I want to get rid of Abigen pub fields. don't think they need to be mutable or do they?

**Comment by DaniPopes** [comment]
They do for the builder, allows direct access to maps

**Comment by mattsse** [comment]
since this is a public type, I'd rather have immutable getter functions.

**Comment by DaniPopes** [comment]
Clippy unrelated, looks like a new lint dropped

**Review by gakonst — APPROVED** [review]

**Pull request merged**